src/DetectChanges/BCBreak/FunctionBased/ParameterDefaultValueChanged.php

```diff
@@ -38,8 +40,23 @@ public function __invoke(
         $changes = Changes::empty();
 
         foreach (Dict\intersect_by_key($fromParametersWithDefaults, $toParametersWithDefaults) as $parameterIndex => $parameter) {
-            $defaultValueFrom = $parameter->getDefaultValue();
-            $defaultValueTo   = $toParametersWithDefaults[$parameterIndex]->getDefaultValue();
+            try {
+                $defaultValueFrom = $parameter->getDefaultValue();
+                $defaultValueTo   = $toParametersWithDefaults[$parameterIndex]->getDefaultValue();
+            } catch (Throwable $throwable) {
+                $parameterDefaultExpression   = $parameter->getDefaultValueExpression();
+                $toParameterDefaultExpression = $toParametersWithDefaults[$parameterIndex]->getDefaultValueExpression();
+
+                if (
+                    $toParameterDefaultExpression instanceof Expr &&
+                    $parameterDefaultExpression instanceof  Expr &&
```

Comment:
This would introduce the php-parser as dep since the use statement. I could switch the check to `!== null` but we are still then relying on php-parser internals so i decided for now to check explicitly on the type.

Reply:
We would probably need php-parser for rendering the AST anyway, no? 🤔

Not happy about the added dependency, but it is what it is.

I will be very conservative on expression support though: don't want this to drift out of control 😁

Reply:
Yeah i can understand. I thought about only checking on `Expr_New` and `Expr_ClassConstFetch` as the type so we dont catch to much. WDYT?

Reply:
Finally having time to getting back to this:

I would also say it is too optimistic, that why i mentioned only catching the 2 types. But what do you mean by rendering it back to text? `getType` seemed to me the only possibilty to get a string like comparsion from `Expr`.

Reply:
> But what do you mean by rendering it back to text?

If the expressions change, we need to render the `Expr` as a `string` for the end-user to understand what has changed.

This should be done with the PHP-Parser PrettyPrinter, I suppose.

Reply:
Added now the prettyprinter and checking on the string retrieved from there. If this is fine from behaviour pov i would try satisfy the SA tools then e.g. getting psalm happy.